### PR TITLE
feat: Change order to return result and response as promises

### PR DESCRIPTION
**Pull request by peterMangrove**

### CODE REVIEW

**Comment by peterMangrove** [comment]
Fixes part of https://github.com/mangrovedao/mangrove-ts/issues/173
Should fix https://github.com/mangrovedao/mangrove-dApp/issues/204

**Review by espendk — CHANGES_REQUESTED** [review]
lgtm! Some smaller comments

File: packages/mangrove.js/src/util/trade.ts [hunk 1]
```diff
@@ -337,6 +365,10 @@ class Trade {
       gives,
       market
     );
+    logger.debug("Market order raw receipt", {
+      contextInfo: "market.marketOrder",
+      data: { receipt: receipt },
+    });
```
[the remark this answers is not in the document]
Comment (peterMangrove) [reply packages/mangrove.js/src/util/trade.ts]: Its a mistake, only one is needed. I have removed one

File: packages/mangrove.js/test/integration/market.integration.test.ts [hunk 2]
```diff
@@ -16,6 +16,7 @@ import * as mockito from "ts-mockito";
 import { Bigish } from "../../dist/nodejs/types";
 import { MgvReader } from "../../src/types/typechain/MgvReader";
 import { Deferred } from "../../src/util";
+import { builtinModules } from "module";
```
[the remark this answers is not in the document]
Comment (peterMangrove) [reply packages/mangrove.js/test/integration/market.integration.test.ts]: No, i have removed the import

**Review by espendk — APPROVED** [review]

**Pull request merged**